scroll-animations-1/Overview.bs

```diff
@@ -369,76 +372,81 @@ spec: cssom-view-1; type: dfn;
 	Animation type: not animatable
 	</pre>
 
-	Specifies attachment behavior for each [=scroll progress timeline=]
-	associated with this element.
+	Specifies the [=name attachment=]
+	of each [=scroll progress timeline=] name
+	declared by this element (see 'scroll-timeline-name').
+
+	As an alternative to attaching locally
+	(to a timeline sourced from this element),
+	a declared name’s corresponding
+	<dfn local-lt=attachment>name attachment</dfn>
+	can defer to a descendant’s timeline
+	(''scroll-timeline-attachment/defer'')
+	or connect this element’s timeline
+	to a matching name declared by and scoped to an ancestor
+	(''scroll-timeline-attachment/ancestor''),
+	allowing its [[#timeline-scope|scope]] to expand
+	beyond this element and its descendants.
 
-	A scroll timeline has a <dfn>attachment</dfn>,
-	that determines which [=scroll container=] and {{ScrollTimeline/axis}} to use
-	when calculating the [progress of the timeline](#scroll-timeline-progress).
+	Values have the following meanings:
 
 	<dl dfn-type=value dfn-for=scroll-timeline-attachment>
 		<dt><dfn>local</dfn>
 		<dd>
-			The [=attachment=] is this timeline.
+			This [=scroll progress timeline=] name
+			[=attaches=] to the corresponding [=scroll progress timeline=]
+			defined by this [=scroll container=].
+
+			If this box is not a [=scroll container=],
+			then the corresponding [=scroll progress timeline=] declaration
+			is ignored.
+
+			<!-- This is the same implication as 'scroll-timeline' not applying to [=scroll containers=],
```

Comment:
Right, but we can't do that for the same reasons (name) attachment can't affect named timeline lookup. :-(

- We need to resolve `animation-timeline` at computed-value time (at the latest), since animations using that timeline (name) can affect those computed values. At that time, the box hasn't been created/updated yet, so we can't make statements about the box.
- It creates a circularity within the same frame, since an animation can affect whether or not an element's box _is_ a scroll container. (Right?)

Luckily we should be able to just say that `scroll-timeline-name` affects "all elements", but that the timeline is inactive if the box isn't a scroll container. I don't think this catastrophically harms any use-cases.

Reply:
We know it's a scroll container by this point tho, right? That doesn't depend on any layout information, just the 'overflow' value.

Reply:
@tabatkins Ah, OK, yes, if checking the computed value of `overflow` is good enough, then that part is fine.

But there's still a circularity, since animations can affect the computed value of `overflow`! I'd _prefer_ to fix that the way I said above, because it keeps timeline lookup more "stable". If we for some reason don't want to do that, then another idea is to look at `overflow` from the "base style" instead (the "base style"  concept doesn't entirely exist yet AFAIK, but it's proposed in https://github.com/w3c/csswg-drafts/pull/6688). That should make it impossible for `scroll-timeline-*` to affect its own applicability.

Reply:
@fantasai How do you think we should deal with that circularity? Any other ideas?

Reply:
Yeah I'm fine with your "all elements, but it's an inactive timeline if it's not scrollable" idea. There's no search involved, so it's predictable anyway.

Reply:
I'm annoyed that we can't make it ignored, since that's what we typically do for properties that don't apply to a particular box type. But since it's an error case, making it easier to implement is also reasonable.